src/nvim/tui/tui.c

```diff
@@ -1575,7 +1576,7 @@ static void augment_terminfo(TUIData *data, const char *term,
   // can use colons like ISO 8613-6:1994/ITU T.416:1993 says.
   bool has_colon_rgb = false
     // per GNOME bug #685759 and bug #704449
-    || (vte_version >= 3600)
+    || (vte_version >= 3600 && !tmux && !screen)
     || iterm || iterm_pretending_xterm  // per analysis of VT100Terminal.m
     // per http://invisible-island.net/xterm/xterm.log.html#xterm_282
     || true_xterm;
```

Comment:
Looking at https://github.com/neovim/neovim/commit/8898793adeb3a82fe50da4258c30940e10ebcc9d#diff-3ac88981a139ac0eee448a6bd19c7569L1539 it seems like `!old_truecolor_env && !rxvt && ...` are also worth including as positive knowledge that `has_colon_rgb=false`.